src/Tribe/Log.php

```diff
@@ -0,0 +1,354 @@
+	 * Schedules a daily cleanup task if one is not already in place.
+	 */
+	protected function register_cleanup_task() {
+		if ( ! wp_next_scheduled( self::CLEANUP ) ) {
+			wp_schedule_event( strtotime( '+1 day' ), 'daily', self::CLEANUP );
+		}
+	}
+
+	/**
+	 * Call the cleanup() method for each available logging engine.
+	 *
+	 * We don't just call it on the current engine since, if there was a recent change,
+	 * we'll generally still want the now unused engine's output to be cleaned up.
+	 */
+	public function do_cleanup() {
+		foreach ( $this->get_logging_engines() as $engine ) {
+			/**
+			 * @var Tribe__Log__Logger $engine
+			 */
+			$engine->cleanup();
+		}
+	}
+
+	/**
+	 * Logs a debug-level entry.
+	 *
+	 * @param string $entry
+	 * @param string $src
+	 */
+	public function log_debug( $entry, $src ) {
+		$this->log( $entry, self::DEBUG, $src );
+	}
+
+	/**
+	 * Logs a warning.
+	 *
+	 * @param string $entry
+	 * @param string $src
+	 */
+	public function log_warning( $entry, $src ) {
+		$this->log( $entry, self::WARNING, $src );
+	}
+
+	/**
+	 * Logs an error.
+	 *
+	 * @param string $entry
+	 * @param string $src
+	 */
+	public function log_error( $entry, $src ) {
+		$this->log( $entry, self::ERROR, $src );
+	}
+
+	/**
+	 * Adds an entry to the log (if it is at the appropriate level, etc).
+	 *
+	 * This is simply a shorthand for calling log() on the current logger.
+	 */
+	public function log( $entry, $type = self::DEBUG, $src = '' ) {
+		if ( $this->should_log( $type ) ) {
+			$this->get_current_logger()->log( $entry, $type, $src );
+		}
+	}
+
+	/**
+	 * Returns a list of available logging engines as an array where each
+	 * key is the classname and the value is the logger itself.
+	 *
+	 * @return array
+	 */
+	public function get_logging_engines() {
+		$available_engines = array();
+		$bundled_engines   = array(
+			'Tribe__Log__File_Logger',
+		);
+
+		foreach ( $bundled_engines as $engine_class ) {
+			$engine = $this->get_engine( $engine_class );
+
+			// Check that we have a valid engine that is available for use in the current environment
+			if ( $engine && $engine->is_available() ) {
+				$available_engines[ $engine_class ] = $engine;
+			}
+		}
+
+		/**
+		 * Offers a chance to modify the array of currently available logging engines.
+		 *
+		 * The array is organized with each key as the class name of the logging
+		 * implementation and the matching value is the actual logger object.
+		 *
+		 * @var array $available_engines
+		 */
+		return apply_filters( 'tribe_common_logging_engines', $available_engines );
+	}
+
+	/**
+	 * Returns the currently active logger or null if none is set/none are
+	 * available.
+	 *
+	 * @return Tribe__Log__Logger|null
+	 */
+	public function get_current_logger() {
+		if ( ! $this->current_logger ) {
+			$engine = tribe_get_option( 'logging_engine', null );
+			$available = $this->get_logging_engines();
+
+			if ( empty( $engine ) || ! isset( $available[ $engine ] ) ) {
+				return null;
```

Comment:
would it not be better to return `false` rather than `null`?
I think it's quite more ablative about failure. Here and elsewhere.
Thinking about testing too.